Services/Common/OJS.Services.Common.Data/Implementations/DataService.cs

```diff
@@ -1,215 +1,226 @@
-namespace OJS.Services.Common.Data.Implementations
+namespace OJS.Services.Common.Data.Implementations;
+
+using Microsoft.EntityFrameworkCore;
+using Microsoft.EntityFrameworkCore.ChangeTracking;
+using OJS.Common.Extensions;
+using OJS.Common.Utils;
+using OJS.Data;
+using OJS.Data.Models.Common;
+using OJS.Services.Common.Models.Users;
+using OJS.Services.Infrastructure.Extensions;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Linq.Expressions;
+using System.Threading.Tasks;
+
+public class DataService<TEntity> : IDataService<TEntity>
+    where TEntity : class, IEntity
 {
-    using Microsoft.EntityFrameworkCore;
-    using Microsoft.EntityFrameworkCore.ChangeTracking;
-    using OJS.Common.Extensions;
-    using OJS.Common.Utils;
-    using OJS.Data;
-    using OJS.Data.Models.Common;
-    using OJS.Services.Common.Models.Users;
-    using OJS.Services.Infrastructure.Extensions;
-    using System;
-    using System.Collections.Generic;
-    using System.Linq;
-    using System.Linq.Expressions;
-    using System.Threading.Tasks;
-
-    public class DataService<TEntity> : IDataService<TEntity>
-        where TEntity : class, IEntity
+    private readonly OjsDbContext db;
+    private readonly DbSet<TEntity> dbSet;
+    private bool ignoreQueryFiltersOnNextQuery;
+    private bool useQueryFiltersOnNextQuery;
+    private bool ignoreQueryFilters;
```

Comment:
We can reuse the `ignoreQueryFiltersOnNextQuery` instead of adding new boolean, by making it nullable.
```
if ((this.ignoreQueryFilters && this.ignoreQueryFiltersOnNextQuery == null) || this.ignoreQueryFiltersOnNextQuery == true)
{
  Ignore()
  this.ignoreQueryFiltersOnNextQuery = null;
}
```

Reply:
While removing one `if` check might seem beneficial, I believe keeping the bool field for enabling `queryFilters` is still necessary. Since the `DataService` class is used in many places, maintaining simplicity in the logic is crucial. In my opinion, adding the extra field enhances readability.

Reply:
To me is somehow confusing to have 2 variables responsible for the same thing. By default both are `false`, but the `useQueryFiltersOnNextQuery` will take precedence (as this is the default behavior of the service). Same goes if both are `true`. But if `useQueryFiltersOnNextQuery == false` and `ignoreQueryFiltersOnNextQuery == true`, the latter will take precedence.

Reply:
Done.